File: task-03/Go/Sub task 4/subtask4.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func printDiamond(output *os.File, n int) {
	if n < 1 {
		fmt.Fprintln(output, "Number should be greater than 1")
		return
	}

	mid := n / 2

	// Upper part of the diamond
	for i := 1; i <= n; i += 2 {
		spaces := mid - i/2
		fmt.Fprintln(output, strings.Repeat(" ", spaces)+strings.Repeat("*", i))
	}

	// Lower part of the diamond
	if n%2 == 0 {
		for i := n - 1; i >= 1; i -= 2 {
			spaces := mid - i/2 + 1
			fmt.Fprintln(output, strings.Repeat(" ", spaces-1)+strings.Repeat("*", i))
		}
	} else {
		for i := n - 2; i >= 1; i -= 2 {
			spaces := mid - i/2
			fmt.Fprintln(output, strings.Repeat(" ", spaces)+strings.Repeat("*", i))
		}
	}
}

func main() {
	inputFile, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Failed to open input.txt:", err)
		return
	}
	defer inputFile.Close()

	outputFile, err := os.Create("output.txt")
	if err != nil {
		fmt.Println("Failed to open output.txt:", err)
		return
	}
	defer outputFile.Close()

	scanner := bufio.NewScanner(inputFile)
	if scanner.Scan() {
		n, err := strconv.Atoi(strings.TrimSpace(scanner.Text()))
		if err != nil {
			fmt.Println("Invalid number:", err)
			return
		}
		printDiamond(outputFile, n)
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading input.txt:", err)
	}
	fmt.Println("File transfer was successful")
}

File: task-03/C++/Sub task 4/subtask4.cpp
#include <iostream>
#include <fstream>
#include <string>

void print_diamond(std::ofstream &output, int n) {
    if (n < 1) {
        output << "Number should be greater than 1\n";
        return;
    }

    int mid = n / 2;

    // Upper part of the diamond
    for (int i = 1; i <= n; i += 2) {
        int spaces = mid - i / 2;
        output << std::string(spaces, ' ') << std::string(i, '*') << "\n";
    }

    // Lower part of the diamond
    if (n % 2 == 0) {
        for (int i = n - 1; i >= 1; i -= 2) {
            int spaces = mid - i / 2 + 1;
            output << std::string(spaces - 1, ' ') << std::string(i, '*') << "\n";
        }
    } else {
        for (int i = n - 2; i >= 1; i -= 2) {
            int spaces = mid - i / 2;
            output << std::string(spaces, ' ') << std::string(i, '*') << "\n";
        }
    }
}

int main() {
    int n;
    std::ifstream input("input.txt");
    if (!input) {
        std::cerr << "Failed to open input.txt\n";
        return 1;
    }

    std::ofstream output("output.txt");
    if (!output) {
        std::cerr << "Failed to open output.txt\n";
        return 1;
    }

    input >> n;
    input.close();

    print_diamond(output, n);
    output.close();
    
    std::cout << "File transfer was successful\n";
    return 0;
}

